### PR TITLE
Retrieve Test tags from local branch changes

### PR DESCRIPTION
**Pull request by diegolovison**
Apply some changes inside `tests/**/*.robot` or `tests/**/*.resource`
Run the script `cd tests && python read_pr.py`
The output:
```
destructive_tests: ...
slow_tests: ...
fast_tests: ...
```

### CODE REVIEW

File: tests/read_pr.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        data = self.get_diff()
+        files_changed = []
+        for line in data:
+            if line.startswith('diff --git '):
+                files_changed.append(line)
+        infos = []
+        for f in files_changed:
+            is_resource = f and f.endswith('.resource')
+            is_robot = f.endswith('.robot')
+            if is_robot or is_resource:
```
Comment (tarukumar) [inline tests/read_pr.py:25]: do we need line 22 23 and 24. line can be be sorted in below code :

`    for f in files_changed:
        if f.endswith(('.robot', '.resource')):
            infos.append(self.get_sections_info(f))`

File: tests/read_pr.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        self.keyword_key = '*** Keywords ***'
+        self.magic_tag_separator = '_______'
+        self.magic_tag = '[Tags]'
+
+    def run(self):
+        data = self.get_diff()
+        files_changed = []
+        for line in data:
+            if line.startswith('diff --git '):
+                files_changed.append(line)
```
Comment (tarukumar) [inline tests/read_pr.py:20]: list comprehension could be good solution as well. 

` files_changed = [line for line in data if line.startswith('diff --git ')]`

File: tests/read_pr.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        infos = []
+        for f in files_changed:
+            is_resource = f and f.endswith('.resource')
```
Comment (tarukumar) [inline tests/read_pr.py:23]: why `f and f.endswith('.resource')` ?

**Comment by tarukumar** [comment]
overall looks good to me, however i would suggest to use list comprehension to simplify the filtering process in the code

File: tests/read_pr.py (hunk 3 above)
Comment (diegolovison) [reply tests/read_pr.py:23]: a mistake

**Comment by diegolovison** [comment]
@tarukumar it is ready for review again. we must squash before merging
Thanks for your review

File: tests/read_pr.py (hunk 1 above)
Comment (diegolovison) [reply tests/read_pr.py]: Done

File: tests/read_pr.py (hunk 2 above)
Comment (diegolovison) [reply tests/read_pr.py]: Done

**Review by tarukumar — APPROVED** [review]
lgtm

**Review by bdattoma — APPROVED** [review]

**Pull request merged**